File: java-api/src/main/java/com/fagerland/javaapi/models/UserEntry.java
package com.fagerland.javaapi.models;

import javax.persistence.*;

@Entity
@Table(name="userEntry")
public class UserEntry {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String username;

    private String hashedPassword;

    public UserEntry() {
    }

    public UserEntry(String username, String hashedPassword) {
        this.username = username;
        this.hashedPassword = hashedPassword;
    }

    public Long getId() {
        return id;
    }

    public String getUsername() {
        return username;
    }

    public String getHashedPassword() {
        return hashedPassword;
    }
}

File: kotlin-maven-api/src/main/kotlin/com/fagerland/brotime/models/UserEntry.kt
package com.fagerland.brotime.models

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name="userEntry")
data class UserEntry(
        var username: String?,
        var hashedPassword: String,
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        var id: Long? = null
)

